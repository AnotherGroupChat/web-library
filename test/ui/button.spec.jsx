import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../src/js/component/ui/button';
import Icon from '../../src/js/component/ui/icon';

it('Renders button', () => {
	let tree = renderer.create(
		<Button>Lorem Foo</Button>
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Renders button with an icon', () => {
	let tree = renderer.create(
		<Button>
			<Icon type="16/folder" />
		</Button>
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it('Renders button with an icon and Text', () => {
	let tree = renderer.create(
		<Button>
			<Icon type="16/folder" /> Foo
		</Button>
	).toJSON();
	expect(tree).toMatchSnapshot();
});


it('Renders active button', () => {
	let tree = renderer.create(
		<Button onClick={ () => {} } active={ true }>
			Ipsum Bar
		</Button>
	).toJSON();
	expect(tree).toMatchSnapshot();
});