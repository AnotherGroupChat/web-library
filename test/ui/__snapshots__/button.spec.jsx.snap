// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders active button 1`] = `
<button
  className="btn active"
  onClick={[Function]}
>
  Ipsum Bar
</button>
`;

exports[`Renders button 1`] = `
<button
  className="btn"
  onClick={[Function]}
>
  Lorem Foo
</button>
`;

exports[`Renders button with an icon 1`] = `
<button
  className="btn btn-icon"
  onClick={[Function]}
>
  <svg
    className="svg-icon icon-folder"
    role="img"
    style={Object {}}
  >
    <use
      viewBox="0 0 512 512"
      xlinkHref="/icons/16/folder.svg#folder"
    />
  </svg>
</button>
`;

exports[`Renders button with an icon and Text 1`] = `
<button
  className="btn btn-icon"
  onClick={[Function]}
>
  <svg
    className="svg-icon icon-folder"
    role="img"
    style={Object {}}
  >
    <use
      viewBox="0 0 512 512"
      xlinkHref="/icons/16/folder.svg#folder"
    />
  </svg>
   Foo
</button>
`;
